pyshamir/shamir.py

```diff
@@ -74,7 +74,11 @@ def split(secret: bytes, parts: int, threshold: int) -> list[bytearray]:
         raise ValueError("Secret must be at least 1 byte long")
 
     # Generate random list of x coordinates
-    x_coordinates = [secrets.randbelow(255) for _ in range(1, 256)]
+    x_coordinates = []
+    while len(x_coordinates) < 255:
+        x = secret.randbelow(255)
```

Comment:
this should be `secrets.randbelow(255)`. The unittests are failing but showing as successful. Need to check that in a separate issue.